src/hooks/useAnimationLoop/useAnimationLoop.test.tsx

```diff
@@ -7,4 +8,101 @@ describe('useAnimationLoop', () => {
       initialProps: undefined,
     });
   });
+
+  it('should not execute the callback function when enabled is not passed', async () => {
+    const spy = jest.fn();
+    renderHook(
+      ({ callback }) => {
+        useAnimationLoop(callback);
+      },
+      {
+        initialProps: {
+          callback: () => {
+            spy();
+          },
+        },
+      },
+    );
+
+    waitFor(() => {
+      expect(spy).toBeCalledTimes(0);
+    });
+  });
+
+  it('should execute the callback function when useAnimationLoop is enabled', async () => {
+    const spy = jest.fn();
+    renderHook(
+      ({ callback, enabled }) => {
+        useAnimationLoop(callback, enabled);
+      },
+      {
+        initialProps: {
+          callback: () => {
+            spy();
+          },
+          enabled: true,
+        },
+      },
+    );
+
+    expect(spy).toBeCalledTimes(0);
+    waitFor(() => {
+      expect(spy).toBeCalled();
+    });
+  });
+
+  it('should execute another callback function when it is passed to useAnimationLoop and not execute previously passed callback function', async () => {
+    const spyFirstRender = jest.fn();
+    const spySecondRender = jest.fn();
+    const { rerender } = renderHook(
+      ({ callback, enabled }) => {
+        useAnimationLoop(callback, enabled);
+      },
+      {
+        initialProps: {
+          callback: () => {
+            spyFirstRender();
+          },
```

Comment:
```suggestion
          callback: spyFirstRender,
```

Should also be fine in this case I believe.